packages/packherd/README.md

```diff
@@ -0,0 +1,140 @@
+## @packages/packherd
+
+Herds all dependencies reachable from an entry and packs them.
```

Comment:
Reading through this readme, I kept thinking: "isn't this a description of vite/webpack"?

Vite in particular already uses esbuild as a backend, and it feels like it should do most of what packherd is doing. Can you walk me through the differences between packherd and vite? What features do we need that vite doesn't provide, and have we considered using vite to provide much of this functionality out-of-the-box and reduce our maintenance burden?

Reply:
There are a few things that are provided by `packherd` that are fairly unique to what we're doing here (or going to be doing once `v8-snapshot` gets moved over as well).

When generating the bundle, we need a way to swap out how the bundle is actually created (https://github.com/cypress-io/cypress/blob/bb89994af7e119236bd263234294d325439abce2/packages/packherd/src/packherd.ts#L25). In the v8 snapshot case, this will utilize `snapbuild` which is our custom version of esbuild that returns warnings that allow us to use our [snapshot doctor](https://github.com/cypress-io/tech-briefs/tree/main/briefs/startup-performance#alternatives-to-existing-tooling) to properly create the snapshot bundle.

In addition to generating a bundle, the other big part of `packherd is the ability to provide the hooks that we need to load our snapshotted exports and definitions (described here: https://github.com/cypress-io/cypress/tree/ryanm/feat/add-packherd-to-monorepo/packages/packherd#loading-bundledsnapshotted-modules-with-packherd).